src/oaklib/implementations/semsimian/semsimian_implementation.py

```diff
@@ -116,3 +117,46 @@ def pairwise_similarity(
         sim.phenodigm_score = math.sqrt(sim.jaccard_similarity * sim.ancestor_information_content)
 
         return sim
+
+    def all_by_all_pairwise_similarity(
+        self,
+        subjects: Iterable[CURIE],
+        objects: Iterable[CURIE],
+        predicates: List[PRED_CURIE] = None,
+        min_jaccard_similarity: Optional[float] = None,
+        min_ancestor_information_content: Optional[float] = None,
+    ) -> Iterator[TermPairwiseSimilarity]:
+        """
+        Compute similarity for all combinations of terms in subsets vs all terms in objects
+
+        :param subjects:
+        :param objects:
+        :param predicates:
+        :return:
+        """
+        objects = list(objects)
+        all_results = self.semsimian.all_by_all_pairwise_similarity(
+            set(subjects), set(objects), set(predicates) if predicates else None
+        )
+        for term1_key, values in all_results.items():
```

Comment:
I don't understand why we're doing python N^2 iteration. Isn't the point of this PR to do the N^2 in Rust?

Reply:
Ah I see, the bit above does the all x all, this is just yielding the results. But there is no threshold, so it will do n^2 object creations